syntax = "proto3";

package ocr;

service OcrService {
  // Client sends one image, gets back the recognized text.
  rpc RecognizeImage (OcrRequest) returns (OcrResponse);
}

message OcrRequest {
  int64 batch_id = 1;          // lets you support multiple batches
  int32 image_index = 2;       // position in the batch / upload order
  string filename = 3;         // original filename (for display/logging)
  bytes image_data = 4;        // raw encoded image (e.g. PNG/JPEG)
}

message OcrResponse {
  int64 batch_id = 1;
  int32 image_index = 2;
  string filename = 3;
  string text = 4;             // OCR result
  int64 processing_time_ms = 5;
  bool success = 6;
  string error_message = 7;    // optional - only when success == false
}
